### ISSUE TITLE
Can't build/install fastnetmon in vyos - code 25600 returned

### ISSUE THREAD

**Issue by jokoyuliantoro** [FastVPSEestiOu/fastnetmon#376]
Hi,
I tried to install fastnetmon in vyos 1.1.5 and it failed when it tried to install the build-essential package with code 25600.
I tried to install build-essential first but there is unmet dependencies which I am not sure if I should force-install it.

Appreciate any helps... 

vyos@vyos-fastnetmon:~$ sudo perl fastnetmon_install.pl
Update package manager cache
Install PF_RING dependencies with package manager
Package 'build-essential' install failed with code 25600
Package 'libnuma-dev' install failed with code 25600
^Cvyos@vyos-fastnetmon:~$
vyos@vyos-fastnetmon:~$
vyos@vyos-fastnetmon:~$ sudo apt-get install -y --force-yes build-essential
Reading package lists... Done
Building dependency tree
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 build-essential : Depends: libc6-dev but it is not going to be installed or
                            libc-dev
                   Depends: g++ (>= 4:4.4.3) but it is not going to be installed
E: Broken packages
vyos@vyos-fastnetmon:~$
vyos@vyos-fastnetmon:~$
vyos@vyos-fastnetmon:~$ show sys image
The system currently has the following image(s) installed:

   1: 1.1.5 (default boot)

vyos@vyos-fastnetmon:~$

**Comment by pavel-odintsov** [comment]
Hello!

Have you tried this guide
https://github.com/FastVPSEestiOu/fastnetmon/blob/master/docs/VyOS_INSTALL.md
?

You could try test binary builds for Debian Squeeze if you still have
issues with compiler. Feel free to ask me if you need help here!

On Wednesday, August 19, 2015, jokoyuliantoro <notifications@github.com>
wrote:

> Hi,
> I tried to install fastnetmon in vyos 1.1.5 and it failed when it tried to
> install the build-essential package with code 25600.
> I tried to install build-essential first but there is unmet dependencies
> which I am not sure if I should force-install it.
>
> Appreciate any helps...
>
> vyos@vyos-fastnetmon:~$ sudo perl fastnetmon_install.pl
> Update package manager cache
> Install PF_RING dependencies with package manager
> Package 'build-essential' install failed with code 25600
> Package 'libnuma-dev' install failed with code 25600
> ^Cvyos@vyos-fastnetmon:~$
> vyos@vyos-fastnetmon:~$
> vyos@vyos-fastnetmon:~$ sudo apt-get install -y --force-yes
> build-essential
> Reading package lists... Done
> Building dependency tree
> Reading state information... Done
> Some packages could not be installed. This may mean that you have
> requested an impossible situation or if you are using the unstable
> distribution that some required packages have not yet been created
> or been moved out of Incoming.
> The following information may help to resolve the situation:
>
> The following packages have unmet dependencies:
> build-essential : Depends: libc6-dev but it is not going to be installed or
> libc-dev
> Depends: g++ (>= 4:4.4.3) but it is not going to be installed
> E: Broken packages
> vyos@vyos-fastnetmon:~$
> vyos@vyos-fastnetmon:~$
> vyos@vyos-fastnetmon:~$ show sys image
> The system currently has the following image(s) installed:
>
> 1: 1.1.5 (default boot)
>
> vyos@vyos-fastnetmon:~$
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/FastVPSEestiOu/fastnetmon/issues/376>.
>


-- 
Sincerely yours, Pavel Odintsov

**Comment by jokoyuliantoro** [comment]
Hi Pavel,

Yes, I followed that VyOS_INSTALL.md to the letter and it failed on the last step where the Perl script was executed.
I don't mind trying binary builds but I was kind of hoping that the installation for VyOS would be straight binary instead of compiling one.
I would like to try to make it work in my VyOS. Probably will do some hacks.

**Comment by pavel-odintsov** [comment]
Hello!

Looks like you have broken package manager.... You could try to reinstall VyOS or you could try binary builds. 

Please wait about 10-20 minutes. I will try to write manual for you. But please keep in mind! I could offer binary builds only for unstable version.

**Comment by pavel-odintsov** [comment]
Hello!

I have prepared this manual for you. It's tested with VyOS 1.1.6

Please execute following commands with ssh on VyOS box.

```bash
sudo su

# Add gpg keys for VyOS
wget http://vyos.net/so3group_maintainers.key
apt-key add ./so3group_maintainers.key

# Add Debian repositories
echo "deb http://httpredir.debian.org/debian/ squeeze-lts main contrib non-free" > /etc/apt/sources.list.d/squeeze_lts.list
echo "deb http://httpredir.debian.org/debian/ squeeze main contrib non-free" >> /etc/apt/sources.list.d/squeeze_lts.list

# Install required package for FastNetMon
apt-get update
apt-get install -y libnuma1

# Disable custom apt repo
rm -f /etc/apt/sources.list.d/squeeze_lts.list
apt-get update

wget http://178.62.227.110/fastnetmon_binary_repository/test_binary_builds/fastnetmon-binary-git-0cfdfd5e2062ad94de24f2f383576ea48e6f3a07-debian-6.0.10-x86_64.tar.gz
tar -xf fastnetmon-binary-git-0cfdfd5e2062ad94de24f2f383576ea48e6f3a07-debian-6.0.10-x86_64.tar.gz -C /opt

# Install example configuration file from upstream version
curl https://raw.githubusercontent.com/FastVPSEestiOu/fastnetmon/master/src/fastnetmon.conf -o /etc/fastnetmon.conf

# Enable init script
curl https://raw.githubusercontent.com/FastVPSEestiOu/fastnetmon/master/src/fastnetmon_init_script_debian_6_7 -o/etc/init.d/fastnetmon
sed -i 's#/usr/sbin/fastnetmon#/opt/fastnetmon/fastnetmon#' /etc/init.d/fastnetmon
chmod +x /etc/init.d/fastnetmon 
insserv fastnetmon

/etc/init.d/fastnetmon start
```

It's enough complex but I will build automatic script for this shortly.

**Comment by jokoyuliantoro** [comment]
Thanks a lot, Pavel.
Unfortunately, the step broke at installing libnumal.
I used httpredir.debian.org which led me to mirror.nus.edu.sg and I notice that there are several packages missing during apt-get update.
I changed to other mirror, ftp.hk.debian.org, and apt-get update went OK.
Please see the logs:

root@vyos-fastnetmon:/home/vyos# rm -f /etc/apt/sources.list.d/squeeze_lts.list
root@vyos-fastnetmon:/home/vyos# echo "deb http://ftp.hk.debian.org/debian/ squeeze-lts main contrib non-free" > /etc/apt/sources.list.d/squeeze_lts.list
root@vyos-fastnetmon:/home/vyos# echo "deb http://ftp.hk.debian.org/debian/ squeeze main contrib non-free" >> /etc/apt/sources.list.d/squeeze_lts.list
root@vyos-fastnetmon:/home/vyos# apt-get update
Get:1 http://ftp.hk.debian.org squeeze-lts Release.gpg [819 B]
Ign http://ftp.hk.debian.org/debian/ squeeze-lts/contrib Translation-en
Ign http://ftp.hk.debian.org/debian/ squeeze-lts/main Translation-en
Get:2 http://packages.vyos.net helium Release.gpg [836 B]
Ign http://ftp.hk.debian.org/debian/ squeeze-lts/non-free Translation-en
Get:3 http://ftp.hk.debian.org squeeze Release.gpg [1655 B]
Ign http://ftp.hk.debian.org/debian/ squeeze/contrib Translation-en
Ign http://ftp.hk.debian.org/debian/ squeeze/main Translation-en
Ign http://packages.vyos.net/vyos/ helium/main Translation-en
Ign http://ftp.hk.debian.org/debian/ squeeze/non-free Translation-en
Get:4 http://ftp.hk.debian.org squeeze-lts Release [31.5 kB]
Get:5 http://ftp.hk.debian.org squeeze Release [96.0 kB]
Hit http://packages.vyos.net helium Release
Get:6 http://ftp.hk.debian.org squeeze-lts/main amd64 Packages [320 kB]
Hit http://packages.vyos.net helium/main amd64 Packages
Get:7 http://ftp.hk.debian.org squeeze-lts/contrib amd64 Packages [1684 B]
Get:8 http://ftp.hk.debian.org squeeze-lts/non-free amd64 Packages [1246 B]
Get:9 http://ftp.hk.debian.org squeeze/main amd64 Packages [8370 kB]
Get:10 http://ftp.hk.debian.org squeeze/contrib amd64 Packages [61.7 kB]
Get:11 http://ftp.hk.debian.org squeeze/non-free amd64 Packages [121 kB]
Fetched 9007 kB in 21s (427 kB/s)
Reading package lists... Done
root@vyos-fastnetmon:/home/vyos#
root@vyos-fastnetmon:/home/vyos# apt-get install -y libnumal
Reading package lists... Done
Building dependency tree
Reading state information... Done
E: Unable to locate package libnumal
root@vyos-fastnetmon:/home/vyos# apt-get install -y libnumal
Reading package lists... Done
Building dependency tree
Reading state information... Done
E: Unable to locate package libnumal
root@vyos-fastnetmon:/home/vyos# show version
Version:      VyOS 1.1.6
Description:  VyOS 1.1.6 (helium)
Copyright:    2015 VyOS maintainers and contributors
Built by:     maintainers@vyos.net
Built on:     Mon Aug 17 03:58:33 UTC 2015
Build ID:     1508170358-a3033d5
System type:  x86 64-bit
Boot via:     image
Hypervisor:   VMware
HW model:     VMware Virtual Platform
HW S/N:       VMware-56 4d 6c 8c 84 53 87 38-63 13 85 a8 52 8f 16 a9
HW UUID:      564D6C8C-8453-8738-6313-85A8528F16A9
Uptime:       00:53:30 up 32 min,  2 users,  load average: 0.08, 0.08, 0.06

root@vyos-fastnetmon:/home/vyos#

**Comment by pavel-odintsov** [comment]
Hello!

You have a mistake with package name. Please use "libnuma1" instead of "libnumal". You could download it here: https://packages.debian.org/squeeze/libnuma1

**Comment by jokoyuliantoro** [comment]
Silly me, simple mistake it is.
The installation succeeded nicely.

Thanks a lot for the helps!

**Comment by pavel-odintsov** [comment]
You are welcome :) If you have any further questions, you could ask me with email pavel.odintsov@gmail.com

**Comment by pavel-odintsov** [comment]
Hello, again! 

This guild could broke VyOS so I have created another test build for it. 

```bash
sudo su

# Add gpg keys for VyOS
wget http://vyos.net/so3group_maintainers.key -O/tmp/so3group_maintainers.key
apt-key add /tmp/so3group_maintainers.key

# Add Debian repositories
echo "deb http://httpredir.debian.org/debian/ squeeze-lts main contrib non-free" > /etc/apt/sources.list.d/squeeze_lts.list
echo "deb http://httpredir.debian.org/debian/ squeeze main contrib non-free" >> /etc/apt/sources.list.d/squeeze_lts.list

# Install required package for FastNetMon
apt-get update
apt-get install -y libnuma1

# Disable custom apt repo
rm -f /etc/apt/sources.list.d/squeeze_lts.list
apt-get update

wget http://178.62.227.110/fastnetmon_binary_repository/test_package_build/fastnetmon-binary-git-0cfdfd5e2062ad94de24f2f383576ea48e6f3a07-debian-6.0.10-x86_64.deb -O/tmp/fastnetmon-binary-git-0cfdfd5e2062ad94de24f2f383576ea48e6f3a07-debian-6.0.10-x86_64.deb

dpkg -i /tmp/fastnetmon-binary-git-0cfdfd5e2062ad94de24f2f383576ea48e6f3a07-debian-6.0.10-x86_64.deb

/etc/init.d/fastnetmon start
```

**Comment by jokoyuliantoro** [comment]
Thanks a lot for the update, Pavel.

Even though it is not related to FastNetMon, I noticed that not all Debian mirror has complete update since httpredir.debian.org will resolve in different mirror site every time and some of the fetching will fail.

I am just able to test again today and it seems the FastNetMon package is gone.

root@vyos:/home/vyos# wget http://178.62.227.110/fastnetmon_binary_repository/test_package_build/fastnetmon-binary-git-0cfdfd5e2062ad94de24f2f383576ea48e6f3a07-debian-6.0.10-x86_64.deb -O/tmp/fastnetmon-binary-git-0cfdfd5e2062ad94de24f2f383576ea48e6f3a07-debian-6.0.10-x86_64.deb
Connecting to 178.62.227.110 (178.62.227.110:80)
wget: server returned error: HTTP/1.1 404 Not Found
root@vyos:/home/vyos#

Thanks!

**Comment by pavel-odintsov** [comment]
Hello!

Please use this manual instead https://github.com/FastVPSEestiOu/fastnetmon/blob/master/docs/PACKAGES_INSTALL.md 

I have built new packages and you could use they in production :)

**Comment by jokoyuliantoro** [comment]
Thanks a lot.

I noticed there is an issue with the Netherlands server when I did this wget:
wget http://178.62.227.110/fastnetmon_binary_repository/test_package_build/fastnetmon-git-447aa5b86bb5a248e310c15a4d5945e72594d6cf-debian-6.0-x86_64.deb

The server suspended the FIN packet for 64 secs.

Please see frame 9153 of the following snapshot.

No.     Time           VLAN Source                Destination           Protocol Length Info
   9151 0.000000000         178.62.227.110        192.168.206.94        HTTP     1322   Continuation or non-HTTP traffic

Frame 9151: 1322 bytes on wire (10576 bits), 1322 bytes captured (10576 bits) on interface 0
Ethernet II, Src: Force10N_3f:7b:66 (00:01:e8:3f:7b:66), Dst: DellPcba_66:76:0d (f0:1f:af:66:76:0d)
Internet Protocol Version 4, Src: 178.62.227.110 (178.62.227.110), Dst: 192.168.206.94 (192.168.206.94)
    Version: 4
    Header length: 20 bytes
    Differentiated Services Field: 0x00 (DSCP 0x00: Default; ECN: 0x00: Not-ECT (Not ECN-Capable Transport))
    Total Length: 1308
    Identification: 0xcc9b (52379)
    Flags: 0x02 (Don't Fragment)
    Fragment offset: 0
    Time to live: 48
    Protocol: TCP (6)
    Header checksum: 0x548c [correct]
    Source: 178.62.227.110 (178.62.227.110)
    Destination: 192.168.206.94 (192.168.206.94)
    [Source GeoIP: Unknown]
    [Destination GeoIP: Unknown]
Transmission Control Protocol
Hypertext Transfer Protocol

No.     Time           VLAN Source                Destination           Protocol Length Info
   9152 0.000012000         192.168.206.94        178.62.227.110        TCP      54     38805 > http [ACK] Seq=3196061276 Ack=1748189953 Win=64240 Len=0

Frame 9152: 54 bytes on wire (432 bits), 54 bytes captured (432 bits) on interface 0
Ethernet II, Src: DellPcba_66:76:0d (f0:1f:af:66:76:0d), Dst: Force10N_3f:7b:66 (00:01:e8:3f:7b:66)
Internet Protocol Version 4, Src: 192.168.206.94 (192.168.206.94), Dst: 178.62.227.110 (178.62.227.110)
    Version: 4
    Header length: 20 bytes
    Differentiated Services Field: 0x00 (DSCP 0x00: Default; ECN: 0x00: Not-ECT (Not ECN-Capable Transport))
    Total Length: 40
    Identification: 0x4843 (18499)
    Flags: 0x02 (Don't Fragment)
    Fragment offset: 0
    Time to live: 128
    Protocol: TCP (6)
    Header checksum: 0x0000 [incorrect, should be 0x8dd8 (may be caused by "IP checksum offload"?)]
    Source: 192.168.206.94 (192.168.206.94)
    Destination: 178.62.227.110 (178.62.227.110)
    [Source GeoIP: Unknown]
    [Destination GeoIP: Unknown]
Transmission Control Protocol

No.     Time           VLAN Source                Destination           Protocol Length Info
   9153 64.856605000        178.62.227.110        192.168.206.94        TCP      60     http > 38805 [FIN, ACK] Seq=1748189953 Ack=3196061276 Win=15872 Len=0

Frame 9153: 60 bytes on wire (480 bits), 60 bytes captured (480 bits) on interface 0
Ethernet II, Src: Force10N_3f:7b:66 (00:01:e8:3f:7b:66), Dst: DellPcba_66:76:0d (f0:1f:af:66:76:0d)
Internet Protocol Version 4, Src: 178.62.227.110 (178.62.227.110), Dst: 192.168.206.94 (192.168.206.94)
    Version: 4
    Header length: 20 bytes
    Differentiated Services Field: 0x00 (DSCP 0x00: Default; ECN: 0x00: Not-ECT (Not ECN-Capable Transport))
    Total Length: 40
    Identification: 0xcc9c (52380)
    Flags: 0x02 (Don't Fragment)
    Fragment offset: 0
    Time to live: 48
    Protocol: TCP (6)
    Header checksum: 0x597f [correct]
    Source: 178.62.227.110 (178.62.227.110)
    Destination: 192.168.206.94 (192.168.206.94)
    [Source GeoIP: Unknown]
    [Destination GeoIP: Unknown]
Transmission Control Protocol
VSS-Monitoring ethernet trailer, Source Port: 0

No.     Time           VLAN Source                Destination           Protocol Length Info
   9154 0.000035000         192.168.206.94        178.62.227.110        TCP      54     38805 > http [ACK] Seq=3196061276 Ack=1748189954 Win=64240 Len=0

Frame 9154: 54 bytes on wire (432 bits), 54 bytes captured (432 bits) on interface 0
Ethernet II, Src: DellPcba_66:76:0d (f0:1f:af:66:76:0d), Dst: Force10N_3f:7b:66 (00:01:e8:3f:7b:66)
Internet Protocol Version 4, Src: 192.168.206.94 (192.168.206.94), Dst: 178.62.227.110 (178.62.227.110)
    Version: 4
    Header length: 20 bytes
    Differentiated Services Field: 0x00 (DSCP 0x00: Default; ECN: 0x00: Not-ECT (Not ECN-Capable Transport))
    Total Length: 40
    Identification: 0x496a (18794)
    Flags: 0x02 (Don't Fragment)
    Fragment offset: 0
    Time to live: 128
    Protocol: TCP (6)
    Header checksum: 0x0000 [incorrect, should be 0x8cb1 (may be caused by "IP checksum offload"?)]
    Source: 192.168.206.94 (192.168.206.94)
    Destination: 178.62.227.110 (178.62.227.110)
    [Source GeoIP: Unknown]
    [Destination GeoIP: Unknown]
Transmission Control Protocol

No.     Time           VLAN Source                Destination           Protocol Length Info
   9155 0.000491000         192.168.206.94        178.62.227.110        TCP      54     38805 > http [FIN, ACK] Seq=3196061276 Ack=1748189954 Win=64240 Len=0

Frame 9155: 54 bytes on wire (432 bits), 54 bytes captured (432 bits) on interface 0
Ethernet II, Src: DellPcba_66:76:0d (f0:1f:af:66:76:0d), Dst: Force10N_3f:7b:66 (00:01:e8:3f:7b:66)
Internet Protocol Version 4, Src: 192.168.206.94 (192.168.206.94), Dst: 178.62.227.110 (178.62.227.110)
    Version: 4
    Header length: 20 bytes
    Differentiated Services Field: 0x00 (DSCP 0x00: Default; ECN: 0x00: Not-ECT (Not ECN-Capable Transport))
    Total Length: 40
    Identification: 0x496b (18795)
    Flags: 0x02 (Don't Fragment)
    Fragment offset: 0
    Time to live: 128
    Protocol: TCP (6)
    Header checksum: 0x0000 [incorrect, should be 0x8cb0 (may be caused by "IP checksum offload"?)]
    Source: 192.168.206.94 (192.168.206.94)
    Destination: 178.62.227.110 (178.62.227.110)
    [Source GeoIP: Unknown]
    [Destination GeoIP: Unknown]
Transmission Control Protocol

No.     Time           VLAN Source                Destination           Protocol Length Info
   9156 0.191610000         178.62.227.110        192.168.206.94        TCP      60     http > 38805 [ACK] Seq=1748189954 Ack=3196061277 Win=15872 Len=0

Frame 9156: 60 bytes on wire (480 bits), 60 bytes captured (480 bits) on interface 0
Ethernet II, Src: Force10N_3f:7b:66 (00:01:e8:3f:7b:66), Dst: DellPcba_66:76:0d (f0:1f:af:66:76:0d)
Internet Protocol Version 4, Src: 178.62.227.110 (178.62.227.110), Dst: 192.168.206.94 (192.168.206.94)
    Version: 4
    Header length: 20 bytes
    Differentiated Services Field: 0x00 (DSCP 0x00: Default; ECN: 0x00: Not-ECT (Not ECN-Capable Transport))
    Total Length: 40
    Identification: 0x4c0e (19470)
    Flags: 0x02 (Don't Fragment)
    Fragment offset: 0
    Time to live: 48
    Protocol: TCP (6)
    Header checksum: 0xda0d [correct]
    Source: 178.62.227.110 (178.62.227.110)
    Destination: 192.168.206.94 (192.168.206.94)
    [Source GeoIP: Unknown]
    [Destination GeoIP: Unknown]
Transmission Control Protocol
VSS-Monitoring ethernet trailer, Source Port: 0

**Comment by pavel-odintsov** [comment]
Thanks for feedback!

I have got similar issue with VyOS wget tool. But I could not reproduce this issue with another tools on another platforms.

So here you could find wget from MacOS trace: https://www.dropbox.com/s/bo8r1ncb2f5txe7/lags_with_fnm_download.png?dl=0

**Comment by pavel-odintsov** [comment]
So, I have installed fresh VyOS in VBox and expect huge retransmission rate for really any download.

**Comment by pavel-odintsov** [comment]
Looks like something wrong with Vbox or VyOS itself.

**Comment by jokoyuliantoro** [comment]
I see.
I am still outside, will try different scenarios to look for the problem
underneath.

Thanks..

**Comment by jokoyuliantoro** [comment]
I tried on my other Linux boxes and I did not encounter similar issue.
The download is still succeeded anyway so I think it is not a major blocking issue.
I also tried to analyze the packet trace on the long delayed finish but I could not find anything suspicious.
I can see that the server delayed the FIN packet but I do not know why the server did that.
The behavior appears in frame 12500 of this link https://www.dropbox.com/s/dmsrrq5j4rswsbz/wget2.pcapng?dl=0.

I will move on testing the FNM itself.

Thanks!
